File: binding/src/network/libqnetworkdiskcache.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qnetworkdiskcache.html
pub const qnetworkdiskcache = struct {

    /// New constructs a new QNetworkDiskCache object.
    pub fn New() QtC.QNetworkDiskCache {
        return qtc.QNetworkDiskCache_new();
    }


    /// New2 constructs a new QNetworkDiskCache object.
    pub fn New2(parent: ?*anyopaque) QtC.QNetworkDiskCache {
        return qtc.QNetworkDiskCache_new2(@ptrCast(parent));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QNetworkDiskCache_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qnetworkdiskcache.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn CacheDirectory(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QNetworkDiskCache_CacheDirectory(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qnetworkdiskcache.CacheDirectory: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetCacheDirectory(self: ?*anyopaque, cacheDir: []const u8) void {
        const cacheDir_str = qtc.libqt_string{
    .len = cacheDir.len,
    .data = cacheDir.ptr,
};
qtc.QNetworkDiskCache_SetCacheDirectory(@ptrCast(self), cacheDir_str);
    }

    pub fn MaximumCacheSize(self: ?*anyopaque, ) i64 {
        return qtc.QNetworkDiskCache_MaximumCacheSize(@ptrCast(self));
    }

    pub fn SetMaximumCacheSize(self: ?*anyopaque, size: i64) void {
        qtc.QNetworkDiskCache_SetMaximumCacheSize(@ptrCast(self), size);
    }

    pub fn CacheSize(self: ?*anyopaque, ) i64 {
        return qtc.QNetworkDiskCache_CacheSize(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnCacheSize(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i64) void {
        qtc.QNetworkDiskCache_OnCacheSize(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseCacheSize(self: ?*anyopaque, ) i64 {
        return qtc.QNetworkDiskCache_QBaseCacheSize(@ptrCast(self));
    }

    pub fn MetaData(self: ?*anyopaque, url: ?*anyopaque) QtC.QNetworkCacheMetaData {
        return qtc.QNetworkDiskCache_MetaData(@ptrCast(self), @ptrCast(url));
    }

    /// Allows for overriding the related default method
    pub fn OnMetaData(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) QtC.QNetworkCacheMetaData) void {
        qtc.QNetworkDiskCache_OnMetaData(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseMetaData(self: ?*anyopaque, url: ?*anyopaque) QtC.QNetworkCacheMetaData {
        return qtc.QNetworkDiskCache_QBaseMetaData(@ptrCast(self), @ptrCast(url));
    }

    pub fn UpdateMetaData(self: ?*anyopaque, metaData: ?*anyopaque) void {
        qtc.QNetworkDiskCache_UpdateMetaData(@ptrCast(self), @ptrCast(metaData));
    }

    /// Allows for overriding the related default method
    pub fn OnUpdateMetaData(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QNetworkDiskCache_OnUpdateMetaData(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseUpdateMetaData(self: ?*anyopaque, metaData: ?*anyopaque) void {
        qtc.QNetworkDiskCache_QBaseUpdateMetaData(@ptrCast(self), @ptrCast(metaData));
    }

    pub fn Data(self: ?*anyopaque, url: ?*anyopaque) QtC.QIODevice {
        return qtc.QNetworkDiskCache_Data(@ptrCast(self), @ptrCast(url));
    }

    /// Allows for overriding the related default method
    pub fn OnData(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) QtC.QIODevice) void {
        qtc.QNetworkDiskCache_OnData(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseData(self: ?*anyopaque, url: ?*anyopaque) QtC.QIODevice {
        return qtc.QNetworkDiskCache_QBaseData(@ptrCast(self), @ptrCast(url));
    }

    pub fn Remove(self: ?*anyopaque, url: ?*anyopaque) bool {
        return qtc.QNetworkDiskCache_Remove(@ptrCast(self), @ptrCast(url));
    }

    /// Allows for overriding the related default method
    pub fn OnRemove(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) bool) void {
        qtc.QNetworkDiskCache_OnRemove(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseRemove(self: ?*anyopaque, url: ?*anyopaque) bool {
        return qtc.QNetworkDiskCache_QBaseRemove(@ptrCast(self), @ptrCast(url));
    }

    pub fn Prepare(self: ?*anyopaque, metaData: ?*anyopaque) QtC.QIODevice {
        return qtc.QNetworkDiskCache_Prepare(@ptrCast(self), @ptrCast(metaData));
    }

    /// Allows for overriding the related default method
    pub fn OnPrepare(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) QtC.QIODevice) void {
        qtc.QNetworkDiskCache_OnPrepare(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBasePrepare(self: ?*anyopaque, metaData: ?*anyopaque) QtC.QIODevice {
        return qtc.QNetworkDiskCache_QBasePrepare(@ptrCast(self), @ptrCast(metaData));
    }

    pub fn Insert(self: ?*anyopaque, device: ?*anyopaque) void {
        qtc.QNetworkDiskCache_Insert(@ptrCast(self), @ptrCast(device));
    }

    /// Allows for overriding the related default method
    pub fn OnInsert(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QNetworkDiskCache_OnInsert(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseInsert(self: ?*anyopaque, device: ?*anyopaque) void {
        qtc.QNetworkDiskCache_QBaseInsert(@ptrCast(self), @ptrCast(device));
    }

    pub fn FileMetaData(self: ?*anyopaque, fileName: []const u8) QtC.QNetworkCacheMetaData {
        const fileName_str = qtc.libqt_string{
    .len = fileName.len,
    .data = fileName.ptr,
};
return qtc.QNetworkDiskCache_FileMetaData(@ptrCast(self), fileName_str);
    }

    pub fn Clear(self: ?*anyopaque, ) void {
        qtc.QNetworkDiskCache_Clear(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnClear(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QNetworkDiskCache_OnClear(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseClear(self: ?*anyopaque, ) void {
        qtc.QNetworkDiskCache_QBaseClear(@ptrCast(self));
    }

    pub fn Expire(self: ?*anyopaque, ) i64 {
        return qtc.QNetworkDiskCache_Expire(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnExpire(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i64) void {
        qtc.QNetworkDiskCache_OnExpire(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseExpire(self: ?*anyopaque, ) i64 {
        return qtc.QNetworkDiskCache_QBaseExpire(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QNetworkDiskCache_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qnetworkdiskcache.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QNetworkDiskCache_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qnetworkdiskcache.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QNetworkDiskCache_Delete(@ptrCast(self));
    }
};

File: binding/src/network/libqnetworkdiskcache.cpp
#include <QAbstractNetworkCache>
#include <QIODevice>
#include <QNetworkCacheMetaData>
#include <QNetworkDiskCache>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QUrl>
#include <qnetworkdiskcache.h>
#include "libqnetworkdiskcache.h"
#include "libqnetworkdiskcache.hxx"

QNetworkDiskCache* QNetworkDiskCache_new() {
	 return new VirtualQNetworkDiskCache();
}

QNetworkDiskCache* QNetworkDiskCache_new2(QObject* parent) {
	 return new VirtualQNetworkDiskCache(parent);
}

libqt_string QNetworkDiskCache_Tr(const char* s) {
	QString _ret = QNetworkDiskCache::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QNetworkDiskCache_CacheDirectory(const QNetworkDiskCache* self) {
	QString _ret = self->cacheDirectory();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QNetworkDiskCache_SetCacheDirectory(QNetworkDiskCache* self, const libqt_string cacheDir) {
	self->setCacheDirectory(QString::fromUtf8(cacheDir.data, cacheDir.len));
}

long long QNetworkDiskCache_MaximumCacheSize(const QNetworkDiskCache* self) {
	return self->maximumCacheSize();
}

void QNetworkDiskCache_SetMaximumCacheSize(QNetworkDiskCache* self, long long size) {
	self->setMaximumCacheSize(size);
}

long long QNetworkDiskCache_CacheSize(const QNetworkDiskCache* self) {
	return self->cacheSize();
}

QNetworkCacheMetaData* QNetworkDiskCache_MetaData(QNetworkDiskCache* self, const QUrl* url) {
	return new QNetworkCacheMetaData(self->metaData(*url));
}

void QNetworkDiskCache_UpdateMetaData(QNetworkDiskCache* self, const QNetworkCacheMetaData* metaData) {
	self->updateMetaData(*metaData);
}

QIODevice* QNetworkDiskCache_Data(QNetworkDiskCache* self, const QUrl* url) {
	return self->data(*url);
}

bool QNetworkDiskCache_Remove(QNetworkDiskCache* self, const QUrl* url) {
	return self->remove(*url);
}

QIODevice* QNetworkDiskCache_Prepare(QNetworkDiskCache* self, const QNetworkCacheMetaData* metaData) {
	return self->prepare(*metaData);
}

void QNetworkDiskCache_Insert(QNetworkDiskCache* self, QIODevice* device) {
	self->insert(device);
}

QNetworkCacheMetaData* QNetworkDiskCache_FileMetaData(const QNetworkDiskCache* self, const libqt_string fileName) {
	return new QNetworkCacheMetaData(self->fileMetaData(QString::fromUtf8(fileName.data, fileName.len)));
}

void QNetworkDiskCache_Clear(QNetworkDiskCache* self) {
	self->clear();
}

libqt_string QNetworkDiskCache_Tr2(const char* s, const char* c) {
	QString _ret = QNetworkDiskCache::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QNetworkDiskCache_Tr3(const char* s, const char* c, int n) {
	QString _ret = QNetworkDiskCache::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
long long QNetworkDiskCache_QBaseCacheSize(const QNetworkDiskCache* self) {
	auto* vqnetworkdiskcache = dynamic_cast<const VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_CacheSize_IsBase(true);
	return vqnetworkdiskcache->cacheSize();
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkDiskCache_OnCacheSize(const QNetworkDiskCache* self, intptr_t slot) {
	auto* vqnetworkdiskcache = dynamic_cast<const VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_CacheSize_Callback(reinterpret_cast<VirtualQNetworkDiskCache::QNetworkDiskCache_CacheSize_Callback>(slot));
}
}

// Base class handler implementation
QNetworkCacheMetaData* QNetworkDiskCache_QBaseMetaData(QNetworkDiskCache* self, const QUrl* url) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_MetaData_IsBase(true);
	return new QNetworkCacheMetaData(vqnetworkdiskcache->metaData(*url));
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkDiskCache_OnMetaData(QNetworkDiskCache* self, intptr_t slot) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_MetaData_Callback(reinterpret_cast<VirtualQNetworkDiskCache::QNetworkDiskCache_MetaData_Callback>(slot));
}
}

// Base class handler implementation
void QNetworkDiskCache_QBaseUpdateMetaData(QNetworkDiskCache* self, const QNetworkCacheMetaData* metaData) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_UpdateMetaData_IsBase(true);
	vqnetworkdiskcache->updateMetaData(*metaData);
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkDiskCache_OnUpdateMetaData(QNetworkDiskCache* self, intptr_t slot) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_UpdateMetaData_Callback(reinterpret_cast<VirtualQNetworkDiskCache::QNetworkDiskCache_UpdateMetaData_Callback>(slot));
}
}

// Base class handler implementation
QIODevice* QNetworkDiskCache_QBaseData(QNetworkDiskCache* self, const QUrl* url) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_Data_IsBase(true);
	return vqnetworkdiskcache->data(*url);
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkDiskCache_OnData(QNetworkDiskCache* self, intptr_t slot) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_Data_Callback(reinterpret_cast<VirtualQNetworkDiskCache::QNetworkDiskCache_Data_Callback>(slot));
}
}

// Base class handler implementation
bool QNetworkDiskCache_QBaseRemove(QNetworkDiskCache* self, const QUrl* url) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_Remove_IsBase(true);
	return vqnetworkdiskcache->remove(*url);
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkDiskCache_OnRemove(QNetworkDiskCache* self, intptr_t slot) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_Remove_Callback(reinterpret_cast<VirtualQNetworkDiskCache::QNetworkDiskCache_Remove_Callback>(slot));
}
}

// Base class handler implementation
QIODevice* QNetworkDiskCache_QBasePrepare(QNetworkDiskCache* self, const QNetworkCacheMetaData* metaData) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_Prepare_IsBase(true);
	return vqnetworkdiskcache->prepare(*metaData);
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkDiskCache_OnPrepare(QNetworkDiskCache* self, intptr_t slot) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_Prepare_Callback(reinterpret_cast<VirtualQNetworkDiskCache::QNetworkDiskCache_Prepare_Callback>(slot));
}
}

// Base class handler implementation
void QNetworkDiskCache_QBaseInsert(QNetworkDiskCache* self, QIODevice* device) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_Insert_IsBase(true);
	vqnetworkdiskcache->insert(device);
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkDiskCache_OnInsert(QNetworkDiskCache* self, intptr_t slot) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_Insert_Callback(reinterpret_cast<VirtualQNetworkDiskCache::QNetworkDiskCache_Insert_Callback>(slot));
}
}

// Base class handler implementation
void QNetworkDiskCache_QBaseClear(QNetworkDiskCache* self) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_Clear_IsBase(true);
	vqnetworkdiskcache->clear();
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkDiskCache_OnClear(QNetworkDiskCache* self, intptr_t slot) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_Clear_Callback(reinterpret_cast<VirtualQNetworkDiskCache::QNetworkDiskCache_Clear_Callback>(slot));
}
}

// Derived class handler implementation
long long QNetworkDiskCache_Expire(QNetworkDiskCache* self) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
	return vqnetworkdiskcache->expire();
	} else {
	return self->QNetworkDiskCache::expire();
}
}

// Base class handler implementation
long long QNetworkDiskCache_QBaseExpire(QNetworkDiskCache* self) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_Expire_IsBase(true);
	return vqnetworkdiskcache->expire();
}
}

// Auxiliary method to allow providing re-implementation
void QNetworkDiskCache_OnExpire(QNetworkDiskCache* self, intptr_t slot) {
	auto* vqnetworkdiskcache = dynamic_cast<VirtualQNetworkDiskCache*>(self);
	if (vqnetworkdiskcache && vqnetworkdiskcache->isVirtualQNetworkDiskCache) {
vqnetworkdiskcache->setQNetworkDiskCache_Expire_Callback(reinterpret_cast<VirtualQNetworkDiskCache::QNetworkDiskCache_Expire_Callback>(slot));
}
}

void QNetworkDiskCache_Delete(QNetworkDiskCache* self) {
    delete self;
}

